src/pages/fa-asset-grp/forms/AssetGroupForm.js

```diff
@@ -0,0 +1,178 @@
+import { Grid } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useForm } from 'src/hooks/form'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { FixedAssetsRepository } from 'src/repositories/FixedAssetsRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SystemRepository } from '../../../repositories/SystemRepository'
+import { ControlContext } from 'src/providers/ControlContext'
+import { DataSets } from 'src/resources/DataSets'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import { MasterSource } from 'src/resources/MasterSource'
+
+const validationSchema = yup.object({
+  reference: yup.string().required(),
+  name: yup.string().required()
+})
+
+const initialValues = {
+  recordId: null,
+  reference: null,
+  name: ''
+}
+
+export default function AssetGroupForm({ labels, maxAccess, recordId }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: FixedAssetsRepository.AssetGroup.page
+  })
+
+  const { formik } = useForm({
+    initialValues,
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema,
+    onSubmit: handleSubmit
+  })
+
+  const handleNumberRangeChange = (event, newValue) => {
+    if (newValue) {
+      formik.setFieldValue('nraId', newValue?.recordId)
```

Comment:
no need for if statement 
`formik.setFieldValue('nraId', newValue?.recordId || null)`